4.queueRange.js

```diff
@@ -29,16 +29,20 @@
 //     return this.array.shift()
 //   }
 // }
-// 
+//
 // Imagine the queue contains -145, 45, 11, 98
 // range(q) should return 243
 
 // Write your function here
-const range = (queue) => {
-
-}
-
-
-
+// const range = (queue) => {
+//   if (queue.isEmpty()) return;
+//   let max = queue.peek();
+//   let min = queue.peek();
+//   while (!queue.isEmpty()) {
+//     max = Math.max(max, queue.dequeue());
+//     min = Math.max(min, queue.enqueue());
+//   }
+//   return max - min;
+// };
```

Comment:
enqueue takes in an argument, but then this would be an infinite loop. Instead you should save the dequeue value and then use that for the min and max. You min should Math.min.